src/farm/library.cairo

```diff
@@ -547,4 +607,134 @@ namespace CarbonableFarmer {
             contract_address=contract_address, index=next, total_supply=total_supply
         );
     }
+
+    func _create_vestings{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
+        total_amount: felt
+    ) {
+        alloc_locals;
+
+        let (starkvest_address) = starkvest_address_.read();
+        let (contract_address) = carbonable_project_address_.read();
+        let (total_supply) = IERC721Enumerable.totalSupply(contract_address=contract_address);
+
+        let zero = Uint256(low=0, high=0);
+        let (is_zero) = uint256_eq(total_supply, zero);
+        if (is_zero == TRUE) {
+            return ();
+        }
```

Comment:
Shouldn't this revert?
Why would we want to create a vesting for a project with no supply?

Reply:
Revert is probably proper indeed, actually I cannot imagine a such case